helix-core/src/main/java/org/apache/helix/task/TaskStateModelFactory.java

```diff
@@ -44,39 +48,29 @@
   private final ScheduledExecutorService _taskExecutor;
   private final ScheduledExecutorService _timerTaskExecutor;
   private ThreadPoolExecutorMonitor _monitor;
-  public final static int TASK_THREADPOOL_SIZE = 40;
 
   public TaskStateModelFactory(HelixManager manager, Map<String, TaskFactory> taskFactoryRegistry) {
```

Comment:
You have some duplicate code in your 2 constructors. Why not keep the original style?
And, I still think merging the init logic in a single method or fewer methods make the code cleaner.

Reply:
I can no longer use the original style because `_taskExecutor` is created differently - the first constructor's logic is no longer a subset of the second constructor's. 

As a result of that, I factored out the timer task thread pool creation etc into methods to avoid duplicate code.

Reply:
I'm not convinced. It seems that if you put the get threadpool size and createTaskExecutor(threadPoolSize) into one method, then you can put this method as a parameter so you can do exactly what was here before.